cmd/cloud-run/rotate-signify-certificate/rotate_signify_certificate.py

```diff
@@ -0,0 +1,278 @@
+"""Simple PubSub handler to rotate signify certificates"""
+
+import datetime
+import os
+import base64
+import json
+import sys
+import tempfile
+import traceback
+from typing import Any, Dict, List
+import requests
+from google.cloud import secretmanager
+from flask import Flask, Response, request, make_response
+from cryptography import x509
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.serialization import pkcs7, Encoding, PrivateFormat
+from cryptography.hazmat.primitives.asymmetric import rsa
+
+app = Flask(__name__)
+project_id: str = os.getenv("PROJECT_ID")
+component_name: str = os.getenv("COMPONENT_NAME", "signify-certificate-rotator")
+application_name: str = os.getenv("APPLICATION_NAME", "secret-rotator")
+
+
+# TODO(kacpermalachowski): Move it to common package
+class LogEntry(dict):
+    """Simplifies logging by returning a JSON string."""
+
+    def __str__(self):
+        return json.dumps(self)
+
+
+@app.route("/", methods=["POST"])
+def rotate_signify_secret() -> Response:
+    """HTTP webhook handler for rotating Signify secrets."""
+    log_fields: Dict[str, Any] = prepare_log_fields()
+    log_fields["labels"]["io.kyma.app"] = "signify-certificate-rotate"
+
+    try:
+        if project_id is None:
+            raise ValueError("Unknown project id")
+
+        pubsub_message = get_pubsub_message()
+
+        secret_rotate_msg = extract_message_data(pubsub_message)
+
+        if secret_rotate_msg["labels"]["type"] != "signify":
```

Comment:
Violation of Open/Closed Principle. hardcoded signify string.

Reply:
Added option to configure it via env var.